### DISCUSSION TITLE
Langchain backend for Agents

### DISCUSSION THREAD

**Discussion by aayushchhabra1999** [microsoft/autogen#62] [Q&A]
We have an existing Langchain-based application but want to migrate to AutoGen. In the technical paper for Autogen, in **Section 5.1 General Guidelines for Using Autogen**, in point 5 (2), it is mentioned that "When existing tools from LangChain etc. are helpful, one can use them as tool-backend for AutoGen agents." - How can we do that? Are there any existing examples? Can someone share a brief on this? I believe this will be helpful to lots of people who would want to migrate from Langchain to AutoGen.

@sonichi @qingyun-wu

**Comment by sonichi** [comment]
Check this: https://github.com/microsoft/autogen/pull/58
If you'd like to review it and share your feedback, it'll be great!

**Comment by mjmjmmmjjj** [comment]
Can be used as a custom function.
